patch/v6.8/out-of-tree-build/0001-ipu7-ipu-bridge-Adding-file-for-out-of-tree-support.patch

```diff
@@ -0,0 +1,1545 @@
+From bcedde5645d57ab2a757b902460b4407aad0260f Mon Sep 17 00:00:00 2001
+From: Jason Chen <jason.z.chen@intel.corp-partner.google.com>
+Date: Sun, 7 Jul 2024 22:31:23 +0800
+Subject: [PATCH] ipu7: ipu-bridge: Adding file for out-of-tree support
+
+The IPU7 bridge name has been changed to use ipu-bridge file, which
+is not included in. Therefore, adding ipu-bridge file from kernel.org
+for Out-of-tree support.
+
+Signed-off-by: Jason Chen <jason.z.chen@intel.corp-partner.google.com>
+---
+ drivers/media/pci/intel/cio2-bridge.c | 526 -----------------
+ drivers/media/pci/intel/cio2-bridge.h | 147 -----
+ drivers/media/pci/intel/ipu-bridge.c  | 817 ++++++++++++++++++++++++++
```

Comment:
I don't get it. Why?

So the out-of-tree build is supposed to build ipu7-drivers outside the kernel tree, not copy-and-compile some in-tree driver out of the kernel tree. Your module should be linking to the existing ipu-bridge kernel module, not duplicate its source and compile it along with other ipu7 source files.

Reply:
Thanks for the review, Vicamo.

While we followed the steps from [README#2](https://github.com/intel/ipu7-drivers?tab=readme-ov-file#2-out-of-tree-build) to apply the patches under `patch/<kernel-version>`, it will apply failed because of missing the source.
That's why I generate the folder for `out-of-tree` to avoid confusing with `in-tree` build.

Reply:
I'm not reviewing your patch. I'm confused about what you're proposing. Because whatever you merge here, it will become a headache to me later.

Both patches under v6.8 are to patch `drivers/media/pci/intel/ipu-bridge.c`, which is certainly in v6.8 kernel source. What's the failure you refer to?

Reply:
If you directly clone the ipu7-driver repo on your device, and then try to apply the [patch](https://github.com/intel/ipu7-drivers/blob/main/patch/v6.8/0001-media-pci-ipu-bridge-Support-ov13b10-camera.patch), the `ipu-bridge.c` cannot be found as I mentioned previously.

Reply:
Why do you apply that patch inside your dkms source folder?